src/Symfony/Component/Form/ButtonBuilder.php

```diff
@@ -49,16 +57,36 @@ class ButtonBuilder implements \IteratorAggregate, FormBuilderInterface
      */
     private $options;
 
+    private $dispatcher;
+
     /**
      * @throws InvalidArgumentException if the name is empty
      */
-    public function __construct(?string $name, array $options = [])
+    public function __construct(?string $name, /*EventDispatcherInterface*/ $dispatcher = [], array $options = [])
     {
+        if (\func_num_args() > 1) {
+            if (!$dispatcher instanceof EventDispatcherInterface) {
+                @trigger_error(sprintf('The "%s" method second argument must be an instance of "%s" since Symfony 4.4. The existing array "$options" argument has been moved to the third position.', __METHOD__, EventDispatcherInterface::class), E_USER_DEPRECATED);
+
+                if (!\is_array($dispatcher)) {
+                    throw new \TypeError(sprintf('The second argument passed to the "%s" method must be an instance of "%s" or an array, "%s" given.', __METHOD__, EventDispatcherInterface::class, \is_object($dispatcher) ? \get_class($dispatcher) : \gettype($dispatcher)));
```

Comment:
```suggestion
                    throw new \TypeError(sprintf('The second argument passed to the "%s" method must be an instance of "%s", "%s" given.', __METHOD__, EventDispatcherInterface::class, \is_object($dispatcher) ? \get_class($dispatcher) : \gettype($dispatcher)));
```

The error message should not hint at the deprecated way of calling this method.